rest-e2e-tests/streams-api.test.ts

```diff
@@ -369,4 +369,71 @@ describe('Streams API', () => {
             })
         })
     })
+
+    describe('DELETE /api/v1/streams/:id', () => {
+        it('happy path', async () => {
+            const stream = await getStreamrClient(streamOwner).createStream({
+                name: 'stream-id-' + Date.now(),
+            })
+            const deleted = await Streamr.api.v1.streams
+                .delete(stream.id)
+                .withAuthenticatedUser(streamOwner)
+                .call()
+            assert.equal(deleted.status, 204)
+            assert.equal(deleted.size, 0)
+        })
+        it('deletes a stream with a permission', async () => {
+            const stream = await getStreamrClient(streamOwner).createStream({
+                name: 'stream-id-' + Date.now(),
+            })
+            const sharePermission = await Streamr.api.v1.streams
+                .grant(stream.id, anonymousUser.address, 'stream_share')
+                .withAuthenticatedUser(streamOwner)
+                .call()
+            assert.equal(sharePermission.status, 200)
+            const deleted = await Streamr.api.v1.streams
+                .delete(stream.id)
+                .withAuthenticatedUser(streamOwner)
+                .call()
+            assert.equal(deleted.status, 204)
+            assert.equal(deleted.size, 0)
+        })
+
+        async function assertStreamsStorageNodeCount(streamId: string, storageNodeCount: number) {
+            const nodes = await Streamr.api.v1.storagenodes
+                .findStorageNodesByStream(streamId)
+                .call()
+            if (nodes.status == 404) {
+                return 0
```

Comment:
Should compare 0 to `storageNodeCount` instead of returning the value?

Reply:
It's bit hacky. I couldn't find an endpoint that returns storage nodes after the stream has been deleted. So calling that endpoint returns 404 after stream has been deleted. Any ideas how to improve this? I guess the correct answer would be to query the DB directly from e2e tests, but it's a bit late for that.

Reply:
> It's bit hacky. I couldn't find an endpoint that returns storage nodes after the stream has been deleted. So calling that endpoint returns 404 after stream has been deleted. Any ideas how to improve this? I guess the correct answer would be to query the DB directly from e2e tests, but it's a bit late for that.

Just a thought, maybe have this method just return a count of storage nodes? I.e.
```ts
async function getStreamsStorageNodeCount(streamId: string): Promise<number> {
    ....
}
```

Then in test something like
```ts
assert.equal(await getStreamsStorageNodeCount('stream-id'), 5)
```  
Dunno?

Reply:
Once the Stream has been deleted I can't use the end point `/api/v1/streams/${streamId}/storageNodes` to find a list of storage nodes, because calling that endpoint with a non-existing `streamId` returns 404. I think I'll remove that confusing semi-"assertion".